spec/browser/test/mock-resource.spec.js

```diff
@@ -363,6 +363,67 @@ describe('Test lib / mock resources', () => {
       })
   })
 
+  it('matches headers', (done) => {
+    mockClient(client)
+      .resource('User')
+      .method('all')
+      .with({ headers: { 'x-custom-header': 'value' } })
+
+    client.User.all({ headers: { 'x-custom-header': 'value' } })
```

Comment:
When headers are specified in the mockClient like this, and our client makes a request is it partial match or exact match on headers that we want? 

What if my client has a few middlewares unrelated to this mock/context that adds extra headers `'foo': 'bar'` (or 40 other headers) then with the exact match approach we have to match all those headers as well, while I might only be interested in testing this particular header.

Meanwhile, the headers matcher function on L#408 is a partial match (it allows the extra `'foo': 'bar'`).

I thought that this was included in your prior discussion with @Nevon and possibly what the `isSuperset()` idea was about.

Example:
`client.User.all({ headers: { 'x-custom-header': 'value', 'foo': 'bar' } })` (fails but is a superset / fulfills all expected headers)